rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,202 @@
+- [Sphinx](https://www.sphinx-doc.org/), a popular documentation generator, known for its [readthedocs](https://readthedocs.org/) pages supports Markdown
+
+#### Why CommonMark instead of another Markdown flavor?
+- CommonMark is very near to having a 1.0 release for a standardized and unambiguous syntax specification for Markdown
+- The popular Sphinx documentation generator [supports CommonMark](https://www.sphinx-doc.org/en/master/usage/markdown.html) (in addition to reStructuredText)
+- GitHub's Markdown is [a strict superset of CommonMark](https://github.blog/2017-03-14-a-formal-spec-for-github-markdown/) and they are committed to having full CommonMark conformance
+
+### reStructuredText
+
+[reStructuredText (reST)](https://en.wikipedia.org/wiki/ReStructuredText) is a file
+format originally developed as part of the Docutils project for documenting the Python language.
+Since then, support was added for reST to Sphinx, a popular tool for documenting (Python) projects, and pandoc.
+
+With Sphinx it is possible to document various languages using the concept of [domains](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html). E.g., if we were to have a format for documenting Nix functions, we could implement a domain in Sphinx, as well as a parser that could parse Nix functions from comments and convert them to the Sphinx domain, as is done currently with the [Nixpkgs library](https://github.com/NixOS/nixpkgs/pull/53055).
+
+Language:
+- [Specification](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html)
+- [Primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html)
+- [Demo](https://docutils.readthedocs.io/en/sphinx-docs/user/rst/demo.html)
+
+Tooling:
+- [Sphinx](https://www.sphinx-doc.org/)
+- [Docutils](https://docutils.sourceforge.io/)
+- [Pandoc](https://pandoc.org/) can convert from/to reST to/from many other formats
+
+Examples of users:
+- Python
+- Linux kernel
+- CMake
+- LLVM
+- Majority of Python packages
+- PowerDNS: ReST with [Sphinx](https://www.sphinx-doc.org/) for
+  manuals, guides and manpages
+
+### Asciidoc
+
+TODO: Short overview
+
+[Demo](https://github.com/opendevise/asciidoc-samples/blob/master/demo.adoc)
+
+Tooling:
+- [Antora](https://antora.org/)
+- [Asciidoctor](https://asciidoctor.org/)
+
+### Texinfo
+
+TODO: Short overview
+
+Powerful, interactive and very nice to use (check out `pinfo`), but harder to write.
+
+### Nix EDSL
+
+With a Nix EDSL, linking to options can become trivial and very natural. Users won't have to learn another language either. Docs could also be written directly next to the thing they document with some convenience functions for annotating values with docs.
+
+### Docbook
+
+TODO: Short overview
+
+[Primer](https://docbook.rocks/)
+
+### Overall Comparisons
+
+| Format | Rendered in GitHub | Adoption | Standardized | Goal |
+| --- | --- | --- | --- | --- |
+| Markdown | Yes | Great among websites | No | Easy to use |
```

Comment:
Is it not standardized if one follows CommonMark?

Reply:
If people have to ask "which markdown standard" it's not standardized imo.
Sure, commonmark's the closest thing to "standard markdown" but you'll probably still end up needing extensions on top of that.

This is only a disadvantage if the other formats under discussion don't have that issue of course.

Reply:
Right, so we could say that Markdown itself is standardized, but the standard is insufficient for our goals (i.e. it doesn't have cross-references).